core/shared/src/test/scala/scalaz/zio/BaseCrossPlatformSpec.scala

```diff
@@ -0,0 +1,51 @@
+package scalaz.zio
+
+import java.util.{ Timer, TimerTask }
+
+import org.specs2.Specification
+import org.specs2.execute.AsResult
+import org.specs2.specification.core.{ AsExecution, Execution }
+import scalaz.zio.internal.PlatformLive
+import scalaz.zio.clock.Clock
+import scalaz.zio.console.Console
+import scalaz.zio.random.Random
+import scalaz.zio.system.System
+
+import scala.concurrent.duration.{ Duration, _ }
+import scala.concurrent.{ ExecutionContext, Future }
+
+abstract class BaseCrossPlatformSpec extends Specification with DefaultRuntime {
+
+  override val Platform = PlatformLive.makeDefault().withReportFailure(_ => ())
+
+  implicit val ec: ExecutionContext = ExecutionContext.Implicits.global
+
+  val DefaultTimeout: Duration = 60.seconds
+  val timer                    = new Timer()
+
+  implicit def zioAsExecution[A: AsResult, R >: Clock with Console with System with Random, E]
+    : AsExecution[ZIO[R, E, A]] =
+    io => Execution.withEnvAsync(_ => runToFutureWithTimeout(io, DefaultTimeout))
+
+  protected def runToFutureWithTimeout[E, R >: Clock with Console with System with Random, A: AsResult](
+    io: ZIO[R, E, A],
+    timeout: Duration
+  ): Future[A] = {
+    val p = scala.concurrent.Promise[A]()
+    val task = new TimerTask {
+      override def run(): Unit =
+        try {
+          p.failure(new Exception("TIMEOUT: " + timeout))
+          ()
+        } catch {
+          case _: Throwable => ()
+        }
+    }
+    timer.schedule(task, timeout.toMillis)
+
+    unsafeRunToFuture(io.mapError(FailedZIO(_))).map(p.success)
+    p.future
+  }
+}
+
+case class FailedZIO[E](e: E) extends Throwable
```

Comment:
Can we use the existing `FiberFailure`?

Reply:
Yes we can.
Although `mapError` is used convert unrestricted `E` to something throwable, to be able to `unsafeRunToFuture`. I intended to have separate Exception for this. Otherwise we can end up having `FiberFailure` wrapping another `FiberFailure`. Perhaps `FailedZIO` name enough is not clear enough.

Reply:
Hmm, I'm pretty sure `FiberFailure` already does what you're looking for. You can use `io.sandbox.mapError(FiberFailure(_))` to get the cause, which will contain all of the information from the failure.

Reply:
Done!